src/lib/react/ReactTypeMacro.hx

```diff
@@ -9,35 +9,9 @@ import haxe.macro.TypeTools;
 class ReactTypeMacro
 {
 	#if macro
-	public static function alterComponentSignatures(inClass:ClassType, fields:Array<Field>):Array<Field>
-	{
-		var propsType:ComplexType = macro :Dynamic;
-		var stateType:ComplexType = macro :Dynamic;
-
-		switch (inClass.superClass)
-		{
-			case {params: params, t: _.toString() => 'react.ReactComponentOf'}:
-				propsType = TypeTools.toComplexType(params[0]);
-				stateType = TypeTools.toComplexType(params[1]);
-
-			default:
-		}
-
-		// Only alter setState signature for non-dynamic states
-		switch (ComplexTypeTools.toType(stateType))
-		{
-			case TType(_) if (!hasSetState(fields)):
-				addSetStateType(fields, inClass, propsType, stateType);
-
-			default:
-		}
-
-		return fields;
-	}
-
 	public static function ensureRenderOverride(inClass:ClassType, fields:Array<Field>):Array<Field>
 	{
-		if (!inClass.isExtern)
+		if (!(inClass.isExtern || inClass.meta.has(':ignore_empty_render')))
```

Comment:
Whoop, sorry, I forgot that if I push to the same branch GH adds it to the PR ^^

I wouldn't want your curiosity left unsatisfied ;)

I'm reworking coconut.react, so that coconut views are true react components and because of how coconut ties into the renderer, I have subclassed `ReactComponent` *twice* with an abstract class that doesn't have a `render` method itself. They usual craziness ...

Reply:
Ahah ok, I tend to avoid subclassing twice ReactComponent and forgot that it was even possible :P I renamed it to [`@:ignoreEmptyRender`](https://github.com/kLabz/haxe-react/commit/0435a282c4d9e911c7b21ec8dfd59c77829c1982) btw.

Reply:
Well, it's explicitly discouraged. What could go wrong? ^^